internal/gitprovider/azure/azure.go

```diff
@@ -181,6 +181,25 @@ func (p *provider) ListPullRequests(
 	return pts, nil
 }
 
+// GetCommitURL implements gitprovider.Interface.
+func (p *provider) GetCommitURL(
+	repoURL string,
+	sha string,
+) (string, error) {
+	normalizedURL := git.NormalizeURL(repoURL)
+
+	parsedURL, err := url.Parse(normalizedURL)
+	if err != nil {
+		return "Unknown", fmt.Errorf("error processing repository URL: %s: %s", repoURL, err)
```

Comment:
It would be more conventional to return an empty string when you're returning an error.

```suggestion
		return "", fmt.Errorf("error processing repository URL: %s: %s", repoURL, err)
```

Applies in a few other places as well.